legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -2479,29 +2479,70 @@ function meta::relational::functions::pureToSqlQuery::processTDSSort(f:FunctionE
+  let sortByFunc = $f.parametersValues->evaluateAndDeactivate()->at(1)->byPassRouterInfo()->cast(@InstanceValue).values->at(0);
+    
+  if($sortByFunc->instanceOf(Path),
+  |
+    let path = $sortByFunc->cast(@Path<Nil,Any|*>);
+    let pathName = 'o_'+$path->buildColumnNameOutOfPath();
+    let joinResult = processPath($path, $pathName, '1', $mainQuery, $vars, ^$state(inFilter=false), $nodeId, $aggFromMap, $context->shift(), $extensions).element->cast(@SelectWithCursor).select;
+
+    let merge = $mainSelect->concatenate($joinResult)->mergeSQLQueryData($nodeId, $state, $context, $extensions);
+
+    ^$operation(
+        currentTreeNode = $mainQuery.currentTreeNode->toOne()->findOneNode($mainQuery.select.data->toOne(), $merge.data->toOne()),
+        positionBeforeLastApplyJoinTreeNode = if ($mainQuery.positionBeforeLastApplyJoinTreeNode->isEmpty(),|[],|$mainQuery.positionBeforeLastApplyJoinTreeNode->toOne()->findOneNode($mainQuery.select.data->toOne(), $merge.data->toOne())),
+        select = ^SelectSQLQuery(
+                    columns = $merge.columns,
+                    data = $merge.data,
+                    leftSideOfFilter = $merge.leftSideOfFilter,
+                    filteringOperation = $merge.filteringOperation,
+                    extraFilteringOperation = $merge.extraFilteringOperation,
+                    orderBy = ^OrderBy(column=$joinResult.columns->at(0)->cast(@Alias).relationalElement, direction=meta::relational::metamodel::SortDirection.ASC)
+                )
+    );,
+  | if($sortByFunc->instanceOf(LambdaFunction),
+      |
+        let leftSidePure = $f.parametersValues->at(0);
+        let leftSideOp = processValueSpecificationReturnPropertyMapping($leftSidePure, $currentPropertyMapping, $operation, $vars, $state, $joinType, $nodeId, $aggFromMap, $context, $extensions)->toOne();
+        let leftSide = $leftSideOp.element->cast(@SelectWithCursor);
+
+        let paramValue = $f.parametersValues->at(1)->match([e:StoreMappingRoutedValueSpecification[1]|$e.value,v:ValueSpecification[1]|$v]);
+        let updatedState = $state->updateFunctionParamScope($paramValue.genericType.typeArguments.rawType->toOne()->cast(@FunctionType),$leftSide);
+        let lambdaFunction = $f->instanceValuesAtParameter(1, $vars, $updatedState.inScopeVars)->at(0)->cast(@FunctionDefinition<Any>);
+        let lambdaFunctionExpression = $lambdaFunction.expressionSequence->at(0);
+        let inScopeVarsWithPlaceholdersState = $lambdaFunction->addPlaceHoldersForLambdaOpenVariables($vars, $updatedState);
+        let embeddedMapping = if($leftSideOp.currentPropertyMapping->isEmpty(), | $currentPropertyMapping, | $leftSideOp.currentPropertyMapping);
+        
+        let rightSide = processValueSpecification($lambdaFunctionExpression, $embeddedMapping, $leftSide, $vars, ^$inScopeVarsWithPlaceholdersState(inFilter=false), JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context->shift(), $extensions)->toOne()->cast(@SelectWithCursor);
```

Comment:
why is inFIlter set to False?

Reply:
rightSide refers to the lambda in sortBy and since the columns present in it won't be part of filteringOperation have to set it to false. It's done in the above existing code as well where we handle paths.